src/test/java/com/orbitz/consul/cache/ConsulCacheTest.java

```diff
@@ -103,24 +97,32 @@ public void testServicesAreUniqueByID() throws Exception {
         client.agentClient().register(8080, 20L, serviceName, serviceId2);
         client.agentClient().pass(serviceId2);
 
-        ServiceHealthCache svHealth = ServiceHealthCache.newCache(healthClient, serviceName);
+        try (ServiceHealthCache svHealth = ServiceHealthCache.newCache(healthClient, serviceName)) {
+            svHealth.start();
+            svHealth.awaitInitialized(3, TimeUnit.SECONDS);
 
-        svHealth.start();
-        svHealth.awaitInitialized(3, TimeUnit.SECONDS);
+            ServiceHealthKey serviceKey1 = getServiceHealthKeyFromCache(svHealth, serviceId, 8080);
+            assertNotNull("Should find service key 1 from serviceHealthCache", serviceKey1);
 
-        Agent agent = client.agentClient().getAgent();
-        Thread.sleep(100);
+            ServiceHealthKey serviceKey2 = getServiceHealthKeyFromCache(svHealth, serviceId2, 8080);
+            assertNotNull("Should find service key 2 from serviceHealthCache", serviceKey2);
 
-        ServiceHealthKey serviceKey1 = ServiceHealthKey.of(serviceId, agent.getDebugConfig().getAdvertiseAddrLAN(), 8080);
-        ServiceHealthKey serviceKey2 = ServiceHealthKey.of(serviceId2, agent.getDebugConfig().getAdvertiseAddrLAN(), 8080);
+            ImmutableMap<ServiceHealthKey, ServiceHealth> healthMap = svHealth.getMap();
+            assertEquals(healthMap.size(), 2);
+            ServiceHealth health =healthMap.get(serviceKey1);
+            ServiceHealth health2 = healthMap.get(serviceKey2);
 
-        ImmutableMap<ServiceHealthKey, ServiceHealth> healthMap = svHealth.getMap();
-        assertEquals(healthMap.size(), 2);
-        ServiceHealth health =healthMap.get(serviceKey1);
-        ServiceHealth health2 = healthMap.get(serviceKey2);
+            assertEquals(serviceId, health.getService().getId());
+            assertEquals(serviceId2, health2.getService().getId());
+        }
+    }
 
-        assertEquals(serviceId, health.getService().getId());
-        assertEquals(serviceId2, health2.getService().getId());
+    private ServiceHealthKey getServiceHealthKeyFromCache(ServiceHealthCache cache, String serviceId, int port) {
+        return cache.getMap().keySet()
+                .stream()
+                .filter(key -> serviceId.equals(key.getServiceId()) && (port == key.getPort()))
+                .findFirst()
+                .orElse(null);
```

Comment:
Cannot we use an `Option` instead of `null`?

Reply:
Done.